### DISCUSSION TITLE
Curvature tensor of GND

### DISCUSSION THREAD

**Discussion by wanpe** [mtex-toolbox/mtex#1813] [Ask Anything] [answered]
Hi Mtex Community,
        when I run the code: 
```matlab
ebsd = ebsd('indexed').gridify;
        kappa = ebsd.curvature
        kappa(2,3)'
```     
  The result is
```       
 ans = curvatureTensor (xyz)
          unit: 1/um     
          rank: 2 (3 x 3)
         NaN NaN NaN
         NaN NaN NaN
         NaN NaN NaN
```     
   Is this situation normal? If not, how should we modify it？
        best wishes!
wan

**Comment by ralfHielscher** [comment]
This is normal. It indicates that you are on some grain boundary and no curvature was computed.

Ralf.

**Reply by wanpe** [reply]
Thank you for your reply.
I followed the tutorial on the official website to execute the code：
DS=locationSystem. bcc (ebsd. CS) has popped up a warning.

warning: Miller indices are converted to hkl 
> In Miller/cat (line 21)
  In vector3d/vertcat (line 4)
  In dislocationSystem (line 61)
  In dislocationSystem.bcc (line 163) 
 
I think this is because my material is a HCP structure. When I choose to run code
dS = dislocationSystem.hcp(ebsd.CS)
Warnings and errors have occurred

Warning: There are no predefined slip systems for hcp yet You might have to define those individually. 
> In slipSystem.hcp (line 135)
In dislocationSystem.hcp (line 167) 
When calling 'slipSystem. hcp', the output parameter 'sS' (which may also include other parameters) was not assigned a value.
Error locationSystem. hcp (line 167)
dS = dislocationSystem(symmetrise(slipSystem.hcp(cs),'antipodal'));
What should I do？
best wishes！
wan.

**Reply by kilir** [reply, marked as answer]
Hi,
here is the hint:
>Warning: There are no predefined slip systems for hcp yet You might have to define those individually.

Have a look here: https://mtex-toolbox.github.io/SlipSystems.html and here https://mtex-toolbox.github.io/slipSystem.slipSystem.html

Cheers,
Rüdiger

**Reply by wanpe** [reply]
Hi
Thank you for your prompt, I will try to define this system.
cheers!
wan
